aws/aws_secrets_manager/aws_scm.go

```diff
@@ -106,18 +191,19 @@ func (a *awsSecretsMgr) GetSecret(
 	}
 	secretOut := make(map[string]interface{})
 	if err := json.Unmarshal([]byte(*secretValueOutput.SecretString), &secretOut); err != nil {
-		return nil, secrets.NoVersion, fmt.Errorf("failed to unmarshal secret data: %v", err)
+		secretOut = make(map[string]interface{})
+		secretOut[secretID] = *secretValueOutput.SecretString
```

Comment:
What is the reason you want to return the SecretString as is, instead of explicitly fail? 
Is your goal to return any secret which was not inserted by this library?